### DISCUSSION TITLE
How to inject a service into servlet filter?

### DISCUSSION THREAD

**Discussion by kishore-pothireddy** [dropwizard/dropwizard#5290] [General]
@Inject in a servlet filter is not working and it always shows null value. How can we do that in dropwizard?

In my case, trying to inject a login service that will validate the token in header.

**Comment by zUniQueX** [comment]
Dropwizard doesn't provide dependency injection for servlet filters. It uses Jersey's HK2 container instead of managing an own DI container. Therefore DI only applies to Jersey's scope.

Servlet filters are executed on the webserver level and therefore outside of Jersey.

I don't see a big problem on using a `ContainerRequestFilter` (which is executed later but enables DI) for the login handling. The `dropwizard-auth` package also operates on those filters to provide authentication. If you don't want to have Jersey resource matching applied, you can add the `@PreMatching` annotation to your `ContainerRequestFilter`.

**Reply by kishore-pothireddy** [reply]
Thanks for your quick response @zUniQueX. As mentioned in [doc](https://www.dropwizard.io/en/latest/manual/core.html#servlet-filters) trying to add a filter to admin servlet.

```
environment.admin().addFilter("FilterName",  AdminFilter.class).addMappingForUrlPatterns(EnumSet.of(DispatcherType.REQUEST), true, "/*");
```
Not sure how to use `ContainerRequestFilter` here.

**Reply by zUniQueX** [reply]
The admin environment doesn't use Jersey at all, using `ContainerRequestFilter` there is not possible. Currently we don't have the possibility to use DI in filters on the admin environment.
